jetstream/engine/engine_api.py

```diff
@@ -124,6 +124,60 @@ def get_result_at_slot(self, slot: int) -> SlotData:
     )
 
 
+class Tokenizer(abc.ABC):
```

Comment:
A  build_tokenizer function is sufficient for engine_api.py.  Could you move Tokenizer out of engine_api.py and create a class tokenizer_ap.py or related class to store Tokenizer?